.github/workflows/ci.yml

```diff
@@ -108,14 +110,32 @@ jobs:
       if: matrix.type.name == 'Debug' && runner.os != 'Windows' # Coverage is already generated on Windows when running tests.
       run: gcovr -r $GITHUB_WORKSPACE -x $GITHUB_WORKSPACE/build/coverage.out -s -f 'src/SFML/.*' -f 'include/SFML/.*' ${{ matrix.platform.gcovr_options }} $GITHUB_WORKSPACE
 
+    - name: Download Codecov Uploader
+      if: matrix.type.name == 'Debug' && github.repository == 'SFML/SFML' && env.HAVE_CODECOV_TOKEN == 'true' # Disable upload in forks and when Codecov token isn't available
+      shell: bash
+      run: |
+        platform=$(echo "${{ runner.os }}" | tr '[:upper:]' '[:lower:]')
+        if [ "$platform" == "windows" ]; then CODECOV_BINARY="codecov.exe"; else CODECOV_BINARY="codecov"; fi
+        curl https://keybase.io/codecovsecurity/pgp_keys.asc | gpg --no-default-keyring --keyring trustedkeys.gpg --import
+        curl -s --remote-name-all "https://uploader.codecov.io/latest/$platform/{$CODECOV_BINARY,$CODECOV_BINARY.SHA256SUM,$CODECOV_BINARY.SHA256SUM.sig}"
+        gpgv "$CODECOV_BINARY.SHA256SUM.sig" "$CODECOV_BINARY.SHA256SUM"
+        if [ "$platform" == "windows" ]; then powershell 'If ($(Compare-Object -ReferenceObject $(($(certUtil -hashfile codecov.exe SHA256)[1], "codecov.exe") -join "  ") -DifferenceObject $(Get-Content codecov.exe.SHA256SUM)).length -eq 0) {echo "codecov: OK"} Else {exit 1}';
+        else shasum -a 256 -c codecov.SHA256SUM; fi
+        chmod +x "$CODECOV_BINARY"
+        echo "CODECOV_BINARY=$CODECOV_BINARY" >> $GITHUB_ENV
```

Comment:
Is this just copied out of the codecov action itself?

Reply:
More or less from [here](https://docs.codecov.com/docs/codecov-uploader)... with the exception of us using bash for most of the heavy lifting on Windows as well.